src/plugins/data/public/index_patterns/fields/field_list.ts

```diff
@@ -72,7 +96,15 @@ export class FieldList extends Array<Field> implements IFieldList {
   };
 
   update = (field: FieldSpec) => {
-    const newField = new Field(this.indexPattern, field, this.shortDotsEnable);
+    const newField = new Field(
+      {
```

Comment:
@sulemanof `Deps` must be passed in the last argument to match other similar places in the `data` plugin

Reply:
I'm confused about this "must be passed" as the last argument, because it doesn't depend on other places, it's just a set of arguments passed in constructor. Is there any benefit in passing it as the last argument? Because in that case we should always explicitly pass down the last `optional` argument `shortDotsEnable` as `undefined` in `Field` constructor.

Reply:
On the other hand, we'll get rid of passing it down at all if we take in account your suggestion [here](https://github.com/elastic/kibana/pull/63473#discussion_r417288439), right ?

Reply:
On each places where we implement that pattern dependencies passes as last argument. 
I prefer to see the same for your changes too. 
And it's not related to my suggestion.

Reply:
I've done it as you suggested, but I've faced the most weird thing I've ever seen:
while interacting with the instance of the `FieldList` class, there is a filter method, which is extended from the native `Array` class, it tries to create a `FieldList` instance again:

![image](https://user-images.githubusercontent.com/31325372/80626607-60db6400-8a57-11ea-9089-c081719216f7.png)

and traps into destructuring problem:

![image](https://user-images.githubusercontent.com/31325372/80626848-b3b51b80-8a57-11ea-9900-ed86a45e87f3.png)

That's why I made these dependencies optional:
![image](https://user-images.githubusercontent.com/31325372/80626989-ec54f500-8a57-11ea-9137-783891e0f72a.png)

I suppose it's based on how javascript engine works under the hood, but it's super weird and not evident in any case, and that's could be a good point to refactor this code to not extend the `Array`.

Reply:
Hm... what if we do it in a little bit different way. I see we create instance of FieldList from 4 or 5 places. 

```
export const createFieldList(dependencies: FieldListDependencies) =>  
  class FieldList extends Array<Field> implements IFieldList {
```

Reply:
@ppisljar  @flash1293 I don't like idea to merge this PR as it cause in that case our public API will be changed twice in version `7.8` just to merge Discover and in version `7.9` to fix it in a right way. 

From the other hand I understand that it's a pretty critical for version 7.8 and we should do all possible to close that task.

My suggestion: apply approach described in #64843 by @spalger and save FieldList in Discovery services.

Reply:
Checked, `window.__kbnBundles__['plugin/data'].IndexPatternFieldList` should help